code/_onclick/hud/interactive_emotes.dm

```diff
@@ -0,0 +1,283 @@
+	interaction.owner = target
+	interaction.initiator = src
+	interaction.register_movement_signals()
+	target.queued_interactions += interaction
+
+	if(target.client && target.hud_used)
+		target.hud_used.update_interactive_emotes()
+
+	interaction.transform = matrix(32, 6, MATRIX_TRANSLATE)
+	animate(interaction, transform = matrix(), time = 2.5, easing = CUBIC_EASING)
+
+	interaction.timer_id = addtimer(CALLBACK(interaction, TYPE_PROC_REF(/atom/movable/screen/interaction, end_interaction), FALSE), interaction.timeout, TIMER_STOPPABLE|TIMER_UNIQUE)
+
+	return TRUE
+
+//Mob interactions
+/atom/movable/screen/interaction
+	name = "high five"
+	desc = "You gonna leave them hanging?"
+	icon = 'icons/mob/screen_alert.dmi'
+	icon_state = "drunk2"	//It looks jolly
+	///Sound filed played when interaction is successful
+	var/interaction_sound = 'sound/effects/snap.ogg'
+	///Who this offer for interaction is being made to
+	var/mob/owner
+	///The mob that initiated the interaction
+	var/mob/initiator
+	///Clear itself after a certain amount of time
+	var/timeout = 10 SECONDS
+	///The reference to the existing timer
+	var/timer_id
+
+/atom/movable/screen/interaction/Initialize(mob/user)
+	. = ..()
+	desc += "\nLeft-click to accept interaction. Right-click or SHIFT + left-click to decline."
+
+///Separate proc to register signals; not on Initialize because owner and initiator are not set yet
+/atom/movable/screen/interaction/proc/register_movement_signals()
+	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(interactees_moved))
+	RegisterSignal(initiator, COMSIG_MOVABLE_MOVED, PROC_REF(interactees_moved))
+
+///What to do when either the owner or the initiating mob moves
+/atom/movable/screen/interaction/proc/interactees_moved()
+	if(!owner.Adjacent(initiator))
+		end_interaction(FALSE)
+
+///Functions to end the interaction
+/atom/movable/screen/interaction/proc/end_interaction(success = TRUE)
+	if(!success)
+		owner.visible_message(failure_message())
+	UnregisterSignal(initiator, COMSIG_MOVABLE_MOVED)
+	UnregisterSignal(owner, COMSIG_MOVABLE_MOVED)
+	qdel(src)
+
+//Delete itself from the owner's hud and list of queued interactions
+/atom/movable/screen/interaction/Destroy()
+	deltimer(timer_id)
+	owner.queued_interactions -= src
+	if(owner.client && owner.hud_used)
+		owner.client.screen -= src
+		owner.hud_used.update_interactive_emotes()
+
+	. = ..()
+
+/atom/movable/screen/interaction/RightClick(mob/user)
+	end_interaction(FALSE)
+
+/atom/movable/screen/interaction/ShiftClick(mob/user)
+	. = ..()
+	end_interaction(FALSE)
+
+/atom/movable/screen/interaction/Click(location, control, params)
+	var/list/modifiers = params2list(params)
+	if(modifiers[SHIFT_CLICK] || modifiers[RIGHT_CLICK])	//These modifiers will deny the interaction
+		return ..()
+
+	if(usr != owner || !can_interact(owner))
+		end_interaction(FALSE)
+		return FALSE
+
+	action()
+
+///Called when the interaction is accepted
+/atom/movable/screen/interaction/proc/action()
```

Comment:
Why does this need to be its own proc? As far as I can see you're only calling it in one place

Reply:
Doesn't need to but good organization to split up the important parts. Someone might want to call this elsewhere.

Reply:
I am going to kill you